### PR TITLE
get ethereum-benchmark-tests running again

### PR DESCRIPTION
**Pull request by rolfyone**
partially addresses #3384

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

**Comment by Nashatyrev** [comment]
There are 3 options to run JMH tests: 
- From IDEA with `jmh` plugin
- With `./gradlew jmh`
- With `java -jar eth-benchmark-tests-develop-jmh.jar ".*"`

@rolfyone what method are you trying to address with this PR?

**Comment by Nashatyrev** [comment]
With IDEA I have the following out: 
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.openjdk.jmh.util.Utils (file:/C:/Users/Admin/.gradle/caches/modules-2/files-2.1/org.openjdk.jmh/jmh-core/1.25/4f0e5e7c894cbc3e34da69db45ed41bc4d29d21e/jmh-core-1.25.jar) to field java.io.PrintStream.charOut
WARNING: Please consider reporting this to the maintainers of org.openjdk.jmh.util.Utils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Exception in thread "main" java.lang.RuntimeException: ERROR: Unable to find the resource: /META-INF/BenchmarkList
	at org.openjdk.jmh.runner.AbstractResourceReader.getReaders(AbstractResourceReader.java:98)
	at org.openjdk.jmh.runner.BenchmarkList.find(BenchmarkList.java:122)
	at org.openjdk.jmh.runner.Runner.internalRun(Runner.java:266)
	at org.openjdk.jmh.runner.Runner.run(Runner.java:209)
	at org.openjdk.jmh.Main.main(Main.java:71)
```

That might be related to #2744. I've made a couple of efforts to fix this but got stuck

**Comment by Nashatyrev** [comment]
When running via `gradle` or JAR I got the following results: 
```
# Run progress: 40,00% complete, ETA 00:00:02
# Fork: 3 of 5
# Warmup Iteration   1: WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
<failure>

java.lang.ExceptionInInitializerError
        at tech.pegasys.teku.bls.BLSSecretKey.fromBytes(BLSSecretKey.java:38)
        at tech.pegasys.teku.bls.BLSSecretKey.fromBytesModR(BLSSecretKey.java:53)
        at tech.pegasys.teku.bls.BLSKeyPair.random(BLSKeyPair.java:45)
        at tech.pegasys.teku.bls.BLSPublicKey.random(BLSPublicKey.java:41)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.lambda$new$0(DataStructureUtil.java:82)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomPublicKeyBytes(DataStructureUtil.java:190)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomValidator(DataStructureUtil.java:576)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.lambda$randomSSZList$1(DataStructureUtil.java:144)
        at java.base/java.util.stream.Streams$RangeLongSpliterator.forEachRemaining(Streams.java:228)
        at java.base/java.util.stream.LongPipeline$Head.forEach(LongPipeline.java:577)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomSSZList(DataStructureUtil.java:144)
        at tech.pegasys.teku.datastructures.util.BeaconStateBuilder.initDefaults(BeaconStateBuilder.java:130)
        at tech.pegasys.teku.datastructures.util.BeaconStateBuilder.<init>(BeaconStateBuilder.java:69)
        at tech.pegasys.teku.datastructures.util.BeaconStateBuilder.create(BeaconStateBuilder.java:76)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomBeaconState(DataStructureUtil.java:607)
        at tech.pegasys.teku.datastructures.util.DataStructureUtil.randomBeaconState(DataStructureUtil.java:599)
        at tech.pegasys.teku.benchmarks.SSZBenchmark.<clinit>(SSZBenchmark.java:27)
        at tech.pegasys.teku.benchmarks.jmh_generated.SSZBenchmark_BeaconStateSerialization_jmhTest._jmh_tryInit_f_sszbenchmark0_0(SSZBenchmark_BeaconStateSerialization_jmhTest.java:333)
        at tech.pegasys.teku.benchmarks.jmh_generated.SSZBenchmark_BeaconStateSerialization_jmhTest.BeaconStateSerialization_Throughput(SSZBenchmark_BeaconStateSerialization_jmhTest.java:71)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:453)
        at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:437)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.UnsupportedOperationException: No class provided, and an appropriate one cannot be found.
        at org.apache.logging.log4j.LogManager.callerClass(LogManager.java:576)
        at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:601)
        at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:588)
        at tech.pegasys.teku.bls.BLS.<clinit>(BLS.java:46)
        ... 31 more
```
This could be locally tweaked by 
```patch
Index: bls/src/main/java/tech/pegasys/teku/bls/BLS.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- bls/src/main/java/tech/pegasys/teku/bls/BLS.java	(revision b5b3c2ef6f5fe0c7e924e99c2cc1826c6943a3f5)
+++ bls/src/main/java/tech/pegasys/teku/bls/BLS.java	(date 1608645980381)
@@ -43,7 +43,7 @@
  */
 public class BLS {
 
-  private static final Logger LOG = LogManager.getLogger();
+  private static final Logger LOG = LogManager.getLogger(BLS.class);
 
   private static BLS12381 BLS_IMPL;
 ```
 
However if any classes with the same logger initialization are created the problem is getting back 
Note the warning: 
```
# Warmup Iteration   1: WARNING: sun.reflect.Reflection.getCallerClass is not supported.
```

**Comment by Nashatyrev** [comment]
And what platform are you testing on?

**Comment by rolfyone** [comment]
The jmh tests were largely not running on mac, presumably others. The 'important' change is the inclusion of bls in the jmh context, because without that very few tests get run.
I still have 1 set of tests not running, which is curious... `ShuffleBenchmark`

I've been mostly testing with `gradle jmh` but it'd be nice if Intellij wasn't unhappy. I'll try that too. Currently my Intellij really doesn't like `TransitionBenchmark` because an abstract class is annotated with the `@State` annotation. I'll downgrade to the old versions and see if that goes away.

Basically I've been running from the `eth-benchmark-tests` folder...

**Review by Nashatyrev — APPROVED** [review]
LGMT! 
From the `eth-benchmark-tests` dir it indeed started working 👍
